src/overlays/actors/ovl_En_Insect/z_en_insect.h

```diff
@@ -17,16 +27,23 @@ typedef struct EnInsect {
     /* 0x01F0 */ Vec3s jointTable[24];
     /* 0x0280 */ Vec3s morphTable[24];
     /* 0x0310 */ EnInsectActionFunc actionFunc;
-    /* 0x0314 */ u16 unk_314;
+    /* 0x0314 */ u16 insectFlags;
     /* 0x0316 */ s16 unk_316;
     /* 0x0318 */ s16 unk_318;
-    /* 0x031A */ s16 unk_31A;
-    /* 0x031C */ s16 unk_31C;
-    /* 0x031E */ s16 unk_31E;
+    /* 0x031A */ s16 actionTimer;
+    /* 0x031C */ s16 lifeTimer;
+    /* 0x031E */ s16 crawlSoundTimer;
     /* 0x0320 */ ObjMakekinsuta* soilActor;
     /* 0x0324 */ f32 unk_324;
     /* 0x0328 */ s16 unk_328;
     /* 0x032A */ u8 unk_32A;
 } EnInsect; // size = 0x032C
 
+typedef enum {
+    /* 0 */ INSECT_PERMANENT,
+    /* 1 */ INSECT_SPAWNED,
+    /* 2 */ INSECT_FIRST_DROPPED,
+    /* 3 */ INSECT_EXTRA_DROPPED // An additional bug spawned by dropping one from a bottle.
+} EnInsectType;
+
```

Comment:
I think this can be done better. 

Type 0 is exclusive to obj_mure, which is an actor that not only spawns a few small creatures a child actors, it can directly manipulate the child spawns to achieve certain behaviors. As I also mentioned in another comment, there is more to Type 0 than not having a life timer. Type 0 should probably have a name tied to it's spawn source, since the unusual behavior is likely done to reduce bg collision calls.

Types 1-3 from what I can tell essentially exhibit the same overall behavior, with the exception being that Type 1 is missing `INSECT_FLAG_ON_GROUND` thus UPDBGCHECKINFO_FLAG_0 isn't set (not sure how this one difference affects things). `INSECT_SPAWNED` isn't as expressive of it's purpose as it could be (it's spawned by bushes and rocks when the player discovers them hiding underneath them), and I don't like `SPAWNED` as it's past tense (which doesn't make sense in the context of ActorSpawn calls). `FIRST_DROPPED` and `EXTRA_DROPPED` could just be named to reflect that these types are only spawned when dropping bugs from a bottle. A better name could eliminate the need for a comment on Type 3

Lastly, you probably should name give the types names like `INSECT_TYPE_*`.